virttest/qemu_vm.py

```diff
@@ -4249,10 +4249,21 @@ def _shell_reboot(session, timeout):
         if self.params.get("mac_changeable") == "yes":
             utils_net.update_mac_ip_address(self)
 
-        if serial:
-            return self.wait_for_serial_login(timeout=(timeout - shutdown_dur))
-        return self.wait_for_login(nic_index=nic_index,
-                                   timeout=(timeout - shutdown_dur))
+        end = time.time() + timeout - shutdown_dur
+        while True:
+            try:
+                _timeout = max(10, end - time.time())
+                if serial:
+                    return self.wait_for_serial_login(timeout=_timeout)
+                return self.wait_for_login(nic_index=nic_index,
+                                           timeout=_timeout)
+            except virt_vm.VMError as details:
```

Comment:
Good point, but I think it should be applied for `wait_for_login` and `wait_for_serial_login` since users can call these two methods during migration as well.

Btw, which types of exceptions did you see? we can only ignore them to reduce overhead.

Reply:
Various. Anyway it depends on the question from https://github.com/avocado-framework/avocado-vt/pull/2289#discussion_r332964279 I'm becoming to like the option no 2 better as that would completely eliminate the need for this (as it'll basically became part of the `get_address` and eventually `login`, if needed).